cpp/src/arrow/compute/light_array.h

```diff
@@ -135,15 +135,33 @@ class ARROW_EXPORT KeyColumnArray {
   /// Only valid if this is a view into a varbinary type
   uint32_t* mutable_offsets() {
     DCHECK(!metadata_.is_fixed_length);
+    DCHECK(metadata_.fixed_length == sizeof(uint32_t));
```

Comment:
I don't know what this classes does and why do we need to change this. Perhaps some explanation?

Reply:
This class is a kind of light-weight Arrow array. While its `{mutable_,}offset()` methods provide access to 32-bit offsets of the array, it's missing similar methods for 64-bit offsets, which occur in `LARGE_{BINARY,STRING}` typed-arrays. I added these methods to enable hashing of `LARGE_{BINARY,STRING}` typed-arrays, which was previously [documented](https://github.com/apache/arrow/pull/13880/files#diff-b9b505df74724cd7a82ea42a90f7e49dd9211a8bedabecf8d0c0171b697ade6fL436) as [unsupported](https://github.com/apache/arrow/pull/13880/files#diff-b9b505df74724cd7a82ea42a90f7e49dd9211a8bedabecf8d0c0171b697ade6fL869).

Reply:
The check here for `metadata_.fixed_length == sizeof(uint32_t)` ensures that the offsets are indeed 32-bit.